astroid/rebuilder.py

```diff
@@ -1086,7 +1087,9 @@ def visit_dict(self, node: ast.Dict, parent: NodeNG) -> nodes.Dict:
             end_col_offset=getattr(node, "end_col_offset", None),
             parent=parent,
         )
-        items = list(self._visit_dict_items(node, parent, newnode))
+        items: list[tuple[SuccessfulInferenceResult, SuccessfulInferenceResult]] = list(
```

Comment:
For now this correct, but at some point we'll likely want to change `_visit_dict_items` as I think it is also related to https://github.com/PyCQA/astroid/issues/1649.

I'm fine with adding this here, but shouldn't we just update that method and "future-proof" it?

Reply:
How would you "future-proof" `_visit_dict_items`? The TreeRebuild doesn't do inference, so the return type should be correct with `Generator[tuple[NodeNG, NodeNG], None, None]`.

The annotation is only necessary since lists are invariant.

Reply:
Ah yeah, it was late and I wasn't thinking right. The inference of `Dict.items` should happen at a later stage.